pkg/apis/cluster/v1alpha1/cluster_types.go

```diff
@@ -46,6 +46,7 @@ type Cluster struct {
 // ClusterSpec defines the desired state of Cluster
 type ClusterSpec struct {
 	// Cluster network configuration
+	// +optional
```

Comment:
it becomes a question whether the API client spec is locked at this point for v1alpha1.
this falls under non-breaking changes, so it seems fine to modify the old config, but it's probably a good idea to clone v1alpha1 into v1alpha2 at this point, setup the roundtrip and start accepting changes there.

a google doc that collects ideas for v1alpha2 changes + a tracking issue that links to it would be nice too.

Reply:
Setting up v1alpha2 is a bit more complicated than just roundtripping with apimachinery. It would require either:
- a fully externalized upgrade path, where the v1alpha1 controllers are shut down, the CRDs are migrated to v1alpha2 and then the v1alpha2 controllers are brought up
- a CRD conversion webhook that is built, deployed, and registered. This requires support from controller-runtime: https://github.com/kubernetes-sigs/controller-runtime/issues/275